src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -659,22 +676,33 @@ class GoogleExtensions(
       petServiceAccount <- createUserPetServiceAccount(samUser, project, samRequestContext)
       petKey <- googleKeyCache.getKey(petServiceAccount)
       serviceAccountCredentials = ServiceAccountCredentials.fromStream(new ByteArrayInputStream(petKey.getBytes()))
-      timeInMinutes = duration.getOrElse(60L)
-      queryParams = urlParamsMap + (requestedByQueryParam -> samUser.email.value)
-      url <- googleStorageService
-        .getSignedBlobUrl(
-          bucket,
-          name,
-          serviceAccountCredentials,
-          expirationTime = timeInMinutes,
-          expirationTimeUnit = TimeUnit.MINUTES,
-          queryParams = queryParams
-        )
-        .compile
-        .lastOrError
+      url <- getSignedUrl(samUser, bucket, name, duration, urlParamsMap, serviceAccountCredentials)
```

Comment:
does the duration make it possible to set the expiration longer than an hour? If so is that a security issue?

Reply:
Google has a max time of 7 days https://cloud.google.com/storage/docs/access-control/signed-urls